src/main/java/bio/terra/service/upgrade/Migrate.java

```diff
@@ -36,42 +31,49 @@
     private static final Logger logger = LoggerFactory.getLogger("bio.terra.service.upgrade");
     private final DataRepoJdbcConfiguration dataRepoJdbcConfiguration;
     private final MigrateConfiguration migrateConfiguration;
-
-    @Autowired
-    private Environment env;
+    private final LiquibaseMigrator liquibaseMigrator;
+    private final Environment env;
 
     @Autowired
     public Migrate(DataRepoJdbcConfiguration dataRepoJdbcConfiguration,
-                   MigrateConfiguration migrateConfiguration) {
+                   MigrateConfiguration migrateConfiguration,
+                   LiquibaseMigrator liquibaseMigrator,
+                   Environment env) {
         this.dataRepoJdbcConfiguration = dataRepoJdbcConfiguration;
         this.migrateConfiguration = migrateConfiguration;
+        this.liquibaseMigrator = liquibaseMigrator;
+        this.env = env;
     }
 
     /**
-     * Run liquibase migration
-     *
+     * Run liquibase migration based on the configuration
      */
     public void migrateDatabase() {
+
         String changesetFile = dataRepoJdbcConfiguration.getChangesetFile();
-        try (Connection connection = dataRepoJdbcConfiguration.getDataSource().getConnection()) {
-            Liquibase liquibase = new Liquibase(changesetFile,
-                new ClassLoaderResourceAccessor(),
-                new JdbcConnection(connection));
+        DataSource dataSource = dataRepoJdbcConfiguration.getDataSource();
 
-            logger.info(String.format("dropAllOnStart is set to %s", migrateConfiguration.getDropAllOnStart()));
-            boolean allowDropAllOnStart = Arrays.stream(env.getActiveProfiles()).anyMatch(env -> env.contains("dev")
-                || env.contains("test") || env.contains("int"));
-            logger.info(String.format("Allow dropAllOnStart is set to %s", allowDropAllOnStart));
+        boolean allowDropAllOnStart = Arrays.stream(env.getActiveProfiles()).anyMatch(env -> env.contains("dev")
+            || env.contains("test") || env.contains("int"));
 
+        logger.info("dropAllOnStart={}; allowDropAllOnstart={}; updateAllOnStart={}",
+            migrateConfiguration.getDropAllOnStart(), allowDropAllOnStart, migrateConfiguration.getUpdateAllOnStart());
+
+        // Two booleans mean 4 cases
+        // - dropAll=false; updateAll=false - do nothing
+        // - dropAll=false; updateAll=true - update
+        // - dropAll=true; updateAll=true OR =false - drop and update
+        // The case of dropAll=true and updateAll=false makes no sense, since it would leave you
```

Comment:
Sounds like two booleans is not the correct model then, a single enum type arg would make more sense. It's outside the scope of this change though.